HLTrigger/special/plugins/HLTCTPPSLocalTrackFilter.cc

```diff
@@ -0,0 +1,234 @@
+  desc.add<int>("maxTracks", -1)
+    ->setComment("maximum number of tracks, if smaller than minTracks it will be ignored");
+  desc.add<int>("maxTracksPerArm", -1)
+    ->setComment("maximum number of tracks per arm of the CTPPS detector, if smaller than minTrackPerArm it will be ignored");
+  desc.add<int>("maxTracksPerPot", -1)
+    ->setComment("maximum number of tracks per roman pot of the CTPPS detector, if negative it will be ignored");
+
+  desc.add<int>("triggerType", trigger::TriggerTrack);
+
+  descriptions.add("hltCTPPSLocalTrackFilter", desc);
+  return;
+}
+
+//
+// destructor and constructor
+//
+HLTCTPPSLocalTrackFilter::~HLTCTPPSLocalTrackFilter()= default;
+
+HLTCTPPSLocalTrackFilter::HLTCTPPSLocalTrackFilter(const edm::ParameterSet& iConfig):
+  HLTFilter(iConfig),
+  pixelLocalTrackInputTag_   (iConfig.getParameter< edm::InputTag > ("pixelLocalTrackInputTag")),
+  stripLocalTrackInputTag_   (iConfig.getParameter< edm::InputTag > ("stripLocalTrackInputTag")),
+  diamondLocalTrackInputTag_ (iConfig.getParameter< edm::InputTag > ("diamondLocalTrackInputTag")),
+  detectorBitset_            (iConfig.getParameter< unsigned int  > ("detectorBitset")),
+  minTracks_                 (iConfig.getParameter< int           > ("minTracks")),
+  minTracksPerArm_           (iConfig.getParameter< int           > ("minTracksPerArm")),
+  maxTracks_                 (iConfig.getParameter< int           > ("maxTracks")),
+  maxTracksPerArm_           (iConfig.getParameter< int           > ("maxTracksPerArm")),
+  maxTracksPerPot_           (iConfig.getParameter< int           > ("maxTracksPerPot")),
+  usePixel_   (false),
+  useStrip_   (false),
+  useDiamond_ (false)
+{
+  if(detectorBitset_ & 1)
+    usePixel_ = true;
+  if(detectorBitset_ & 2)
+    useStrip_ = true;
+  if(detectorBitset_ & 4)
+    useDiamond_ = true;
+
+  if(usePixel_)
+    pixelLocalTrackToken_   = consumes<edm::DetSetVector<CTPPSPixelLocalTrack>>(pixelLocalTrackInputTag_);
+  if(useStrip_)
+    stripLocalTrackToken_   = consumes<edm::DetSetVector<TotemRPLocalTrack>>(stripLocalTrackInputTag_);
+  if(useDiamond_)
+    diamondLocalTrackToken_ = consumes<edm::DetSetVector<CTPPSDiamondLocalTrack>>(diamondLocalTrackInputTag_);
+
+  LogDebug("") << "HLTCTPPSLocalTrackFilter: pixelTag/stripTag/diamondTag/bitset/minTracks/minTracksPerArm/maxTracks/maxTracksPerArm/maxTracksPerPot : "
+               << pixelLocalTrackInputTag_.encode() << " "
+               << stripLocalTrackInputTag_.encode() << " "
+               << diamondLocalTrackInputTag_.encode() << " "
+               << detectorBitset_ << " "
+               << minTracks_ << " "
+               << minTracksPerArm_ << " "
+               << maxTracks_ << " "
+               << maxTracksPerArm_ << " "
+               << maxTracksPerPot_;
+}
+
+//
+// member functions
+//
+bool HLTCTPPSLocalTrackFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSetup, trigger::TriggerFilterObjectWithRefs& filterproduct) const
+{
```

Comment:
You do not really make use of the HLTfilter functionality, so then it is better to use a standard EDFilter as base class.